### ISSUE TITLE
Program compiles but fails to execute

### ISSUE THREAD

**Issue by alexspurling** [onyx-lang/onyx#63]
The following code compiles successfully on the latest compiler version (0.1.9 932a6360) but when running the program simply exits with an error code of 1. There are no error messages shown even in verbose mode:

```
use core {println}
use core.iter
use core.string

main :: () {
    iter.as_iter(string.split("hello\nthere", '\n')) |>
        iter.map((g) => println(g));
}
```

Running this in the Try Onyx playground causes an error "CompileError: wasm validation error: at offset 59234: popping value from empty stack".

**Comment by brendanfh** [comment]
Looks like there is an issue when the mapping function returns `void`, like `println` does. This should probably either error or it should be handled properly

**Comment by brendanfh** [comment]
This issue has been resolved with a better compiler error in #75.
